### PR TITLE
Add support for media player assumed state

### PR DESCRIPTION
**Pull request by thecode**
## Breaking change

The frontend media player card for integrations which sets the `assumed_state` to `True` and `state` to  `STATE_ON`/ `STATE_PLAYING`/`STATE_PAUSED`/ `STATE_IDLE` will change from:
image
to:
assumed_state
The example assumes the following supported features are set: `SUPPORT_PREVIOUS_TRACK`, `SUPPORT_PLAY`, `SUPPORT_PAUSE`, `SUPPORT_STOP`, `SUPPORT_NEXT_TRACK`, if one of them is not set the corresponding button will not be displayed.

Currently there is one core integration which sets the `assumed_state` to `True`: [xiaomi_tv](https://github.com/home-assistant/core/blob/f5829173db5b56d7f52465440e340fcea1668b6c/homeassistant/components/xiaomi_tv/media_player.py#L85), but it does not set any of the required supported features to cause the media card to change.

**Custom integrations which meets these conditions will also have the frontend card changed.**
## Proposed change

Discussed in https://github.com/home-assistant/architecture/discussions/720, Add support for media player integrations which does not have a feedback from the device for the current playing mode (playing/paused/idle), if the integration sets the `assumed_state` to true the frontend will display all supported features in case the current state is `STATE_ON`/ `STATE_PLAYING`/`STATE_PAUSED`/ `STATE_IDLE`.

Current behavior:
| state	               | supported features  | assumed state | action                 | icon             |
| ------------------- | ---------------------- | ---------------- | -----------------  | ---------------- |
| playing/paused    | SUPPORT_PREVIOUS_TRACK | false        | media_previous_track   | mdiSkipPrevious  |
| playing                 | SUPPORT_PAUSE/STOP     | false        | media_pause+media_stop | mdiPause+mdiStop | 
| paused/idle          | SUPPORT_PLAY           | false        | media_play             | mdiPlay          |
| on                         | SUPPORT_PLAY/PAUSE     | false        | media_play             | mdiPlayPause     |
| playing/paused    | SUPPORT_NEXT_TRACK     | false        | media_next_track       | mdiSkipNext      |

New options when `assumed_state` is `True` and state is `on/playing/idle/paused`:

| supported features       | assumed state | action                 | icon             |
| ---------------------- | ------------ | ---------------------- | ---------------- |
| SUPPORT_PREVIOUS_TRACK | true         | media_previous_track   | mdiSkipPrevious  |
| SUPPORT_PLAY           | true         | media_play             | mdiPlay          | 
| SUPPORT_PAUSE          | true         | media_pause            | mdiPause         |
| SUPPORT_STOP           | true         | media_stop             | mdiStop          |
| SUPPORT_NEXT_TRACK     | true         | media_next_track       | mdiSkipNext      |

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [x] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion: https://github.com/home-assistant/architecture/discussions/720
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by spacegaier** [comment]
Also, there should be a few more coding places where the logic needs to be changed such as the media player row.

**Comment by thecode** [comment]
> Do we need to show both pause and stop? IIRC the current coding is giving one of the two precedence over the other. Does that have to change with the new `assumed_state` logic or couldn't we keep that part to reduce screen space usage?

This implementation does add an additional button to the media player card, I tried to think of a way to keep it smaller, but any other option would still require the core integration to implement logic which is not aligned with the device capabilities or just drop one of the options.

File: src/data/media-player.ts [hunk 1]
```diff
@@ -261,8 +261,11 @@ export const computeMediaControls = (
     });
   }
 
+  const assumedState =
+    state === "on" && stateObj.attributes.assumed_state === true;
```
[the remark this answers is not in the document]
Comment (balloob) [reply src/data/media-player.ts:265]: I think that would be ok.

**Review by balloob — DISMISSED** [review]

**Comment by thecode** [comment]
> The logic looks good if Elupus comment is integrated.
> 
> Spacegaier is right, we should ensure all places are either using this media controls method or are updated. Looks like just 1 spot left:
> 
> * [x]  state card `state-card-media_player.js`: doesn't show controls
> * [ ]  lovelace entity row `hui-media-player-entity-row.ts`: needs updating. See line 114
> * [x]  ha-bar-media-player `ha-bar-media-player.ts`: Uses `computeMediaControls`
> * [x]  media player card `hui-media-control-card.ts`: Uses `computeMediaControls`

Added assumed state in
* [x]  lovelace entity row `hui-media-player-entity-row.ts`

And the comment from @elupus (assumed state behave the same on any sate which is not unavailable/unknown/off)

**Review by balloob — APPROVED** [review]

**Pull request merged**